controllers/volsync/vshandler.go

```diff
@@ -2030,3 +2078,76 @@ func (v *VSHandler) deleteSnapshot(ctx context.Context,
 
 	return nil
 }
+
+func (v *VSHandler) getRD(pvcName string) (*volsyncv1alpha1.ReplicationDestination, error) {
+	l := v.log.WithValues("pvcName", pvcName)
+
+	// Get RD instance
+	rdInst := &volsyncv1alpha1.ReplicationDestination{}
+
+	err := v.client.Get(v.ctx,
+		types.NamespacedName{
+			Name:      getReplicationDestinationName(pvcName),
+			Namespace: v.owner.GetNamespace(),
+		}, rdInst)
+	if err != nil {
+		if !kerrors.IsNotFound(err) {
+			l.Error(err, "Failed to get ReplicationDestination")
+
+			return nil, fmt.Errorf("error getting replicationdestination (%w)", err)
+		}
+
+		l.Info("No ReplicationDestination found")
+
+		return nil, nil
+	}
+
+	return rdInst, nil
+}
+
+func (v *VSHandler) pauseRD(rdName string) (*volsyncv1alpha1.ReplicationDestination, error) {
+	rd, err := v.getRD(rdName)
+	if err != nil || rd == nil {
+		return nil, err
+	}
+
+	rd.Spec.Paused = true
```

Comment:
This is very minor, but if the RD is already paused, should we skip the updateResource?